### PR TITLE
Fix: Parsing fix for param with arrays and properties (fixes #111)

### LINKED ISSUE #111: `@param {string} employees[].name` is parsed incorrectly
This example comes from http://usejsdoc.org/tags-param.html#parameters-with-properties

Expected result:

```
{
  title: "param",
  name: "employees[].name",
  description: undefined,
  type: ...
}
```

Actual result:

```
{
  title: "param",
  name: "employees",
  description: "[].bar",
  type: ...
}
```

### PR DESCRIPTION
**Pull request by gyandeeps**
@nzakas I am not fully convinced with my approach. Need more set of eyes to evaluate this change.
Still in learning phase when it comes to code for this project.

### CODE REVIEW

**Comment by nzakas** [comment]
I'm not entirely sure either...@constellation, do you have a second to look this over quickly?

File: lib/doctrine.js [hunk 1]
```diff
@@ -324,6 +324,10 @@
                     name += scanIdentifier(last);
 
                 }
+                if(source.charCodeAt(index) === 0x5B  /* '[' */ && source.charCodeAt(index + 1) === 0x5D  /* ']' */){
+                    name += advance();
+                    name += advance();
+                }
                 while (source.charCodeAt(index) === 0x2E  /* '.' */ ||
```
Comment (mysticatea) [inline lib/doctrine.js]: This loop seems to handle nesting `MemberExpression`.
So I think the code for `[]` should be inside of this loop.

Comment (nzakas) [inline lib/doctrine.js]: I tried moving the code inside of the loop, but it broke a couple of tests. I think this is a good start.

**Comment by nzakas** [comment]
LGTM for now. We can always fix it if it turns out to be a problem later.

**Pull request merged**